### PR TITLE
Chore/network layer

### PR DESCRIPTION
**Pull request by mltheus**
## What have changed?

NetWork layer
## Kind of pr:

- [ ] BugFix
- [ ] Feature
- [x] Improvement

## How to test:

Use the app and trigger all possible requests

## Extras

Many classes has a extra 's' in the end, that’s because the name without the s is the old layer, after we launch and see that everything is working as it should we delete the old layer and rename those classes

### CODE REVIEW

File: MercadoPagoSDK/MercadoPagoSDK/RequestLayer/RequestStruct/Requesting.swift [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+protocol RequestProtocol {
+    associatedtype
+        Target: RequestInfos
+    func requestObject<Model: Codable>(model: Model.Type, _ target: Target, completionHandler: @escaping (_ result: Model?, _ error: Error?) -> Void)
+    func requestData(target: Target, completionHandler: @escaping (_ data: Data?, _ error: Error?) -> Void)
+}
+
+final class Requesting<Target: RequestInfos> : RequestProtocol {
+    let MP_DEFAULT_PRODUCT_ID = "BJEO9TFBF6RG01IIIOU0"
```
Comment (jscaramal-meli) [inline MercadoPagoSDK/MercadoPagoSDK/RequestLayer/RequestStruct/Requesting.swift:18]: it's ok to let this here ?

Comment (mltheus) [reply MercadoPagoSDK/MercadoPagoSDK/RequestLayer/RequestStruct/Requesting.swift:18]: Fair point, i kind of copy that from old code but if i am not wrong there is one somewhere else

**Review by jscaramal-meli — APPROVED** [review]

**Review by ViniciusAndradeS — APPROVED** [review]

File: MercadoPagoSDK/MercadoPagoSDK/RequestLayer/RequestStruct/ParameterEncode.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        guard let parameters = parameters,
+            let httpMethod = HTTPMethod(rawValue: urlRequest.httpMethod ?? "GET"),
+            (httpMethod == .get || httpMethod == .delete),
+            var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false) else { return urlRequest }
+
+        var components: [(String, String)] = []
+
+        for key in parameters.keys.sorted(by: <) {
+            let value = parameters[key]!
```
Comment (ViniciusAndradeS) [inline MercadoPagoSDK/MercadoPagoSDK/RequestLayer/RequestStruct/ParameterEncode.swift:28]: Esse force pode dar problema?

File: MercadoPagoSDK/MercadoPagoSDK/Core/PXNavigationHandler.swift [hunk 3]
```diff
@@ -66,13 +66,15 @@ internal class PXNavigationHandler: NSObject {
...
-            self.currentLoadingView?.modalTransitionStyle = .crossDissolve
-            self.currentLoadingView!.dismiss(animated: animated, completion: {
-                self.currentLoadingView = nil
-                if let callback = finishCallback {
-                    callback()
-                }
-            })
+            DispatchQueue.main.async {
+                self.currentLoadingView?.modalTransitionStyle = .crossDissolve
+                self.currentLoadingView!.dismiss(animated: animated, completion: {
```
Comment (ViniciusAndradeS) [inline MercadoPagoSDK/MercadoPagoSDK/Core/PXNavigationHandler.swift:71]: Esse force pode dar problema? Eu sei que já estava ai mas só pra garantir

**Pull request merged**